### PR TITLE
Refactor + some fixes and testing

### PR DESCRIPTION
**Pull request by artursouza**

### CODE REVIEW

File: LICENSE-robfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Copyright (C) 2012 Rob Figueiredo
```
Comment (cicoyle) [inline LICENSE-robfig:1]:
```suggestion
Copyright (C) 2024 Diagrid Inc.
```

Please update the file name as well

File: README.md [hunk 2]
```diff
@@ -79,11 +82,15 @@ Pre-requisites to run the tests locally:
...
+go test -timeout 300s --race ./...
 ```
 
 ## History
 
 This is a fork of [https://github.com/Scalingo/go-etcd-cron](https://github.com/Scalingo/go-etcd-cron), which had been based on [https://github.com/robfig/cron](https://github.com/robfig/cron).
 
-This fork has similar but still different goals from Scalingo's go-etcd-cron library.
\ No newline at end of file
+This fork has different goals from Scalingo's go-etcd-cron library.
```
Comment (cicoyle) [inline README.md:96]: Might be worth adding more details here, like what our diff goals are.

File: cron.go [hunk 3]
```diff
@@ -122,13 +106,13 @@ func WithErrorsHandler(f func(context.Context, Job, error)) CronOpt {
 	})
 }
 
-func WithEtcdMutexBuilder(b EtcdMutexBuilder) CronOpt {
+func WithEtcdClient(c *etcdclient.Client) CronOpt {
```
Comment (cicoyle) [inline cron.go:109]: Im now relying on this func: `WithEtcdMutexBuilder` in scheduler code:
```
	c, err := etcdcron.NewEtcdMutexBuilder(clientv3.Config{Endpoints: etcdEndpoints})
```
Once we switch to this library in scheduler code we will need to keep this change in mind.

**Review by cicoyle — APPROVED** [review]
Small nits, otherwise generally LGTM

File: LICENSE-robfig (hunk 1 above)
Comment (artursouza) [reply LICENSE-robfig:1]: Sorry, wrong change.

File: cron.go (hunk 3 above)
Comment (artursouza) [reply cron.go:109]: Yes, this is cleaner IMO. Compilation errors are the best way to remember to change :)

**Pull request merged**